src/infra/core/src/testing/dataset_test_helper.rs

```diff
@@ -74,6 +78,91 @@ impl DatasetTestHelper {
             })
             .collect()
     }
+
+    pub async fn append_random_data(
+        dataset_repo: &dyn DatasetRepository,
+        dataset_ref: impl Into<DatasetRef>,
+        is_large_file: bool,
```

Comment:
Pass a byte size instead of this boolean, since this is not a choice of algorithm.
The only difference between "small" and "large", is the size of the byte buffer.
Maybe you could declare 2 typical sizes as constants, saying "LARGE_SIZE", "SMALL_SIZE".